docs/source/index.rst

```diff
@@ -52,11 +52,11 @@ such topics as:
    :caption: Arrow Libraries
 
    status
-   C++ <cpp/index>
-   Python <python/index>
-   Java <https://arrow.apache.org/docs/java/>
    C/GLib <https://arrow.apache.org/docs/c_glib/>
+   C++ <cpp/index>
+   Java <java/index>
```

Comment:
Given that C++ and Java are the most complete and mature implementations for me, I think it makes sense to emphasize them.

Reply:
Sure, but I'd prefer to emphasize them in other ways/places. A (seemingly) random sort order in this list of libraries is a bad UX and doesn't convey the relative robustness of the items in the list.